packages/@o3r/core/src/store/types.ts

```diff
@@ -65,27 +65,32 @@ export interface FailActionPayload<T> {
   error?: T;
 }
 
-// eslint-disable-next-line @typescript-eslint/naming-convention
-export type keep<T, K extends keyof T> = (Partial<T> & { [p in K]: T[p] });
+export type Keep<T, K extends keyof T> = (Partial<T> & { [p in K]: T[p] });
```

Comment:
isn't it basically just a `Pick` ?

Reply:
I don't want to refactor the code in this PR
I think Pick is equivalent to `type Pick = { [p in K]: T[p] }`
so here we keep all the fields but some becomes non mandatory but if you defined them you still have to match the initial typing

Reply:
yes I read too fast
```suggestion
export type Keep<T, K extends keyof T> = Partial<T> & Pick<T,K>;
```